crates/evolve/src/rpc/txpool.rs

```diff
@@ -48,31 +48,31 @@ where
     /// Returns a Geth-style `TxpoolContent` with raw RLP hex strings.
     async fn get_txs(&self) -> RpcResult<Vec<Bytes>> {
         //------------------------------------------------------------------//
-        // 1. Iterate pending txs and stop once we hit the byte cap         //
+        // 1. Iterate best txs (sorted by priority) and stop once we hit    //
+        //    the byte cap                                                   //
         //------------------------------------------------------------------//
         let mut total = 0u64;
-        let mut pending_map: Vec<Bytes> = Vec::new();
+        let mut selected_txs: Vec<Bytes> = Vec::new();
 
-        for arc_tx in self.pool.pending_transactions() {
-            // deref Arc<ValidPoolTransaction<_>>
-            let pooled: &ValidPoolTransaction<_> = &arc_tx;
-
-            let sz = pooled.encoded_length() as u64;
+        // Use best_transactions() which returns an iterator of transactions
+        // ordered by their priority (gas price/priority fee)
+        for best_tx in self.pool.best_transactions() {
+            let sz = best_tx.encoded_length() as u64;
```

Comment:
I think we may be able to save an iteration here as well as in ev-node, so total of two iterations if we just use another method that wraps around best_transactions and does this entire calculation over a list of transactions, in addition to the checks in ev-node's caller `GetTxs`.

can be as a follow up

Reply:
we dont do any iteration in ev-node, how would we save an iteration here?

Reply:
https://github.com/evstack/ev-node/blob/04637e6ee86b8a9f22440322b77e7775ce2d8dab/execution/evm/execution.go#L134-L142

was referring this iteration, we go over the list of txs in `ev-node`. If we just do all these checks during the initial iteration, it helps. Albeit, I think this would only be relevant if we're in the 10^5-10^6 range of txs per block which I don't think we're at yet.

Reply:
lets open an issue, for right now i dont think its an issue

Reply:
https://github.com/evstack/ev-reth/issues/31